server/channels/api4/status.go

```diff
@@ -105,8 +107,15 @@ func updateUserStatus(c *Context, w http.ResponseWriter, r *http.Request) {
 	}
 
 	currentStatus, err := c.App.GetStatus(c.Params.UserId)
-	if err == nil && currentStatus.Status == model.StatusOutOfOffice && status.Status != model.StatusOutOfOffice {
-		c.App.DisableAutoResponder(c.AppContext, c.Params.UserId, c.IsSystemAdmin())
+	if err != nil {
+		c.Logger.Warn("Failed to get current status", mlog.Err(err))
+	}
+	if currentStatus.Status == model.StatusOutOfOffice && status.Status != model.StatusOutOfOffice {
+		err = c.App.DisableAutoResponder(c.AppContext, c.Params.UserId, c.IsSystemAdmin())
+		if err != nil {
+			c.Err = err
+			return
+		}
 	}
```

Comment:
Let's wrap this in an `else` as I'm not sure what the default return value of `GetStatus` is
```suggestion
	}
	} else if currentStatus.Status == model.StatusOutOfOffice && status.Status != model.StatusOutOfOffice {
		err = c.App.DisableAutoResponder(c.AppContext, c.Params.UserId, c.IsSystemAdmin())
		if err != nil {
			c.Err = err
			return
		}
	}
```

Reply:
Committed directly as the suggestion